storm-client/src/jvm/org/apache/storm/security/auth/authorizer/SimpleACLAuthorizer.java

```diff
@@ -157,6 +162,25 @@ public boolean permit(ReqContext context, String operation, Map<String, Object>
             }
 
             if (checkUserGroupAllowed(userGroups, topoGroups)) return true;
+
+            if (_topoReadOnlyCommands.contains(operation)){
```

Comment:
The two code blocks L150-L164 and L167-L182 are duplicated except some parameters. Could we extract this to a new method?